dora/core/server/worker/src/main/java/alluxio/worker/modules/BlockWorkerModule.java

```diff
@@ -99,6 +100,10 @@ protected void configure() {
         throw new UnsupportedOperationException("Unsupported block store type.");
     }
 
-    bind(Worker.class).to(DefaultBlockWorker.class).in(Scopes.SINGLETON);
+    if (!Configuration.getBoolean(PropertyKey.WORKER_REGISTER_TO_ALL_MASTERS)) {
```

Comment:
@JiamingMai PTAL if this is okay. Thanks!

Reply:
LGTM